Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -549,11 +549,23 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
         self.init(fromPointer: &oid_t)
     }
 
-    /// Initializes an `ObjectId` from the provided `String`.
+    /// Initializes an `ObjectId` from the provided `String`. Assumes that the given string is a valid ObjectId.
+    /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
     public init(fromString oid: String) {
         self.oid = oid
     }
 
+    /// Initializes an `ObjectId` from the provided `String`. Returns `nil` if the string is not a valid
+    /// ObjectId.
+    /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
+    public init?(ifValid oid: String) {
```

Comment:
If I understand correctly, the only thing that differentiates the failable and non-failable initializer is the argument name. Is that a common convention?

Reply:
they need to be different in *at least* this way for the compiler to be able to distinguish between them. for a single parameter init, there isn't much else we can do to differentiate